sdk/trace/evictedqueue.go

```diff
@@ -29,6 +29,22 @@ func newEvictedQueue[T any](capacity int) evictedQueue[T] {
 	}
 }
 
+func newEvictedQueueEvent(capacity int) evictedQueue[Event] {
```

Comment:
Shouldn't we remove the `newEvictedQueue[T]` method?

Reply:
This method is also used in unit tests. Can be used to help test other methods.

Reply:
We can switch the tests to use the two new methods rather than the generic one.

Reply:
@dmathieu Done. PTAL.